include/User.class.php

```diff
@@ -1545,6 +1545,170 @@ public static function validateHomepage($homepage)
             throw new UserException(_h("Homepage is to long."), ErrorType::VALIDATE_HOMEPAGE_LONG);
         }
     }
+
+    /**
+     * Get nth top player based on ranking scores (called by everyone)
+     *
+     * @param int $ntop
+     *
+     * @return list of a player info with rank
+     */
+    public static function getTopPlayersFromRanking($ntop)
+    {
+        try
+        {
+            $result = DBConnection::get()->query(
+                "SELECT
+                    FIND_IN_SET(scores,
+                        (SELECT GROUP_CONCAT(DISTINCT scores ORDER BY scores DESC)
+                        FROM `{DB_VERSION}_rankings`)
+                    ) AS rank,
+                    id,
+                    username,
+                    scores,
+                    max_scores,
+                    num_races_done
+                FROM `{DB_VERSION}_rankings`
+                ORDER BY rank LIMIT :ntop",
+                DBConnection::FETCH_ALL,
+                [':ntop' => $ntop],
+                [':ntop' => DBConnection::PARAM_INT]
+            );
+        }
+        catch (DBException $e)
+        {
+            Debug::addException($e);
+            return null;
+        }
+        if (!$result)
+        {
+            return null;
+        }
+        return $result;
+    }
+
+    /**
+     * Get ranking of a player (called by everyone)
+     *
+     * @param int $id
+     *
+     * @return player info with rank
+     */
+    public static function getRanking($id)
+    {
+        try
+        {
+            $result = DBConnection::get()->query(
+                "SELECT
+                    FIND_IN_SET(scores,
+                        (SELECT GROUP_CONCAT(DISTINCT scores ORDER BY scores DESC)
+                        FROM `{DB_VERSION}_rankings`)
+                    ) AS rank,
+                    id,
+                    username,
+                    scores,
+                    max_scores,
+                    num_races_done
+                FROM `{DB_VERSION}_rankings`
+                WHERE `id` = :id",
+                DBConnection::FETCH_FIRST,
+                [':id' => $id],
+                [':id' => DBConnection::PARAM_INT]
+            );
+        }
+        catch (DBException $e)
+        {
+            Debug::addException($e);
+            // Initial scores for all users:
+            return
+                [
+                    'rank' => -1,
```

Comment:
This seems to be a default return type. Make a method that returns this instead of duplicating code.